.github/workflows/unit_tests.yml

```diff
@@ -12,41 +12,51 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest]
-        python-version: [3.8, 3.9]
+        os: [ubuntu-22.04,macos-14]
+        python-version: [3.8,3.9]
 
     steps:
       - uses: actions/checkout@v4
-      - uses: conda-incubator/setup-miniconda@v3
-        with:
-          activate-environment: test
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        id: cp310
```

Comment:
what is cp310?

Reply:
It's an id to refer the step in further steps. We dont need this anymore, I removed it.